File: TelerikUIExamplesInCSharp/Examples/Chart/ChartTypes/StockDataPoint.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	public class StockDataPoint: TKChartFinancialDataPoint
	{
		public static List<StockDataPoint> LoadStockPoints(int maxItems)
		{
			List<StockDataPoint> stockPoints = new List<StockDataPoint> ();

			string filePath = NSBundle.MainBundle.PathForResource ("AppleStockPrices", "json");
			NSData json = NSData.FromFile (filePath);
			NSError error = new NSError ();
			NSArray data = (NSArray)NSJsonSerialization.Deserialize (json, NSJsonReadingOptions.AllowFragments, out error);
			NSDateFormatter formatter = new NSDateFormatter ();
			formatter.DateFormat = "dd-MM-yyyy";

			for (int i = 0; i < (int)data.Count; i++) {
				if (i == maxItems) {
					break;
				}
				NSDictionary jsonPoint = data.GetItem<NSDictionary> ((nuint)i);
				StockDataPoint dataPoint = new StockDataPoint ();
				dataPoint.DataXValue = formatter.Parse ( (NSString)jsonPoint ["date"]);
				dataPoint.Open = (NSNumber)jsonPoint ["open"];
				dataPoint.High = (NSNumber)jsonPoint ["high"];
				dataPoint.Low = (NSNumber)jsonPoint ["low"];
				dataPoint.Close = (NSNumber)jsonPoint ["close"];
				dataPoint.Volume = (NSNumber)jsonPoint ["volume"];
				stockPoints.Add (dataPoint);
			}

			return stockPoints;
		}
	}
}

File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/StockDataPoint.swift
//
//  StockDataPoint.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class StockDataPoint: TKChartFinancialDataPoint/*, NSCoding*/ {
    
    class func stockPoints() -> NSArray {
        let dataPoints = NSMutableArray()
        let filePath = NSBundle.mainBundle().pathForResource("AppleStockPrices", ofType: "json")
        let json = NSData(contentsOfFile: filePath!)!
        do {
            let data = try NSJSONSerialization.JSONObjectWithData(json, options: NSJSONReadingOptions.AllowFragments) as! NSArray
            let formatter = NSDateFormatter()
            formatter.dateFormat = "dd-MM-yyyy"
            
            for jsonPoint in data {
                let datapoint = StockDataPoint()
                datapoint.dataXValue = formatter.dateFromString(jsonPoint["date"] as! String)
                datapoint.setOpen(jsonPoint["open"] as! NSNumber)
                datapoint.setHigh(jsonPoint["high"] as! NSNumber)
                datapoint.setLow(jsonPoint["low"] as! NSNumber)
                datapoint.setClose(jsonPoint["close"] as! NSNumber)
                datapoint.setVolume(jsonPoint["volume"] as! NSNumber)
                dataPoints.addObject(datapoint)
            }
        }
        catch {
            
        }
        
        return dataPoints
    }
}
